### PR TITLE
Upload build: allow option `parent-revision`

### PR DESCRIPTION
**Pull request by OliverJAsh**
# Problem

Explain the problem that this pull request aims to resolve.

# Solution

Explain your approach. Sometimes it helps to justify your approach against some others that you didn't choose to explain why yours is better.

# TODO

- [ ] 🤓 Add & update tests (always try to _increase_ test coverage)
- [ ] 🔬 Ensure CI is passing (`yarn lint:ci`, `yarn test`, `yarn tsc:ci`)
- [ ] 📖 Update relevant documentation

### CODE REVIEW

File: docs/docs/guides/contributing.md [hunk 1]
```diff
@@ -153,8 +153,8 @@ Build Tracker is written in Typescript throughout all workspaces. Please do not
 To run the development environment, some shortcuts are provided in the `package.json` scripts:
 
 - `yarn dev` Run the server and application in a hot-reloadable environment using pre-seeded builds on the filesystem
-- `yarn dev:maria` (coming soon) Run the server using a local MariaDB instance
-- `yarn dev:postgres` (coming soon) Run the server using a local Postgres instance
```
Comment (OliverJAsh) [inline docs/docs/guides/contributing.md]: Unrelated change: AFAICS, these exist now!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1 +1 @@
-Please refer to the contributing guidelines at https://buildtracker.dev/docs/contributing
```
Comment (OliverJAsh) [inline CONTRIBUTING.md]: Unrelated change: fix broken link

**Comment by montogeek** [comment]
What it your use case? It is different than selecting different revisions in the UI?

**Comment by OliverJAsh** [comment]
I'm just using it for testing now, but if `create-build` has this option, shouldn't `upload-build` also have it? We're already spreading it through from `upload-build` to `create-build`—this PR is just adding the option to the whitelist.

**Comment by paularmstrong** [comment]
There are better ways to list out pass-through args, but it would require a big overhaul. This seems maintainable for now, so let's just go with it. We can copypasta any others over as needed if you see any.
